### PR TITLE
add runtime filter session variable

### PR DESCRIPTION
**Pull request by elsa0520**
link: pingcap/tidb#7275

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -3682,6 +3682,24 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
 
 - 需要注意的是修改该变量不会对已保存的老数据产生影响，只会对修改变量后的新写入数据使用对应版本格式保存。
 
+### `tidb_runtime_filter_mode`
+
```
Comment (winoros) [inline system-variables.md:3686]:
```suggestion
> **警告：**
>
> 该功能为实验特性，不建议在生产环境中使用。
```

File: system-variables.md [hunk 2]
```diff
@@ -3682,6 +3682,26 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+> 该功能为实验特性，不建议在生产环境中使用。
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值: `OFF`
+- 可选值: `OFF`, `LOCAL`
+- 该变量用于控制 Runtime Filter 功能的模式。从 v7.2 开始引入，目前为实验性功能。`OFF` 指的是不开启 Runtime Filter 功能，`LOCAL` 指的开启并使用 LOCAL 模式的 Runtime Filter 功能。
+
+### `tidb_runtime_filter_type`
+
```
Comment (elsa0520) [inline system-variables.md:3697]:
```suggestion
**警告：**
```

File: system-variables.md [hunk 3]
```diff
@@ -3682,6 +3682,26 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+> 该功能为实验特性，不建议在生产环境中使用。
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值: `OFF`
+- 可选值: `OFF`, `LOCAL`
+- 该变量用于控制 Runtime Filter 功能的模式。从 v7.2 开始引入，目前为实验性功能。`OFF` 指的是不开启 Runtime Filter 功能，`LOCAL` 指的开启并使用 LOCAL 模式的 Runtime Filter 功能。
+
+### `tidb_runtime_filter_type`
+**警告：**
```
Comment (elsa0520) [inline system-variables.md:3697]: > **警告：**
>
> 该功能为实验特性，不建议在生产环境中使用。

Comment (elsa0520) [inline system-variables.md:3697]:
```suggestion
> **警告：**
>
> 该功能为实验特性，不建议在生产环境中使用。
```

**Review by winoros — APPROVED** [review]

File: system-variables.md [hunk 4]
```diff
@@ -3682,6 +3682,28 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
 
 - 需要注意的是修改该变量不会对已保存的老数据产生影响，只会对修改变量后的新写入数据使用对应版本格式保存。
 
+### `tidb_runtime_filter_mode`
+> **警告：**
+>
+> 该功能为实验特性，不建议在生产环境中使用。
```
Comment (qiancai) [inline system-variables.md:3688]:
```suggestion
> 该功能为实验特性，不建议在生产环境中使用。

```

File: system-variables.md [hunk 5]
```diff
@@ -3682,6 +3682,28 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值: `OFF`
+- 可选值: `OFF`, `LOCAL`
```
Comment (qiancai) [inline system-variables.md:3693]: Runtime Filter 功能具体是什么功能呢？ 可以说明下吗

File: system-variables.md [hunk 6]
```diff
@@ -3682,6 +3682,28 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+- 该变量用于控制 Runtime Filter 功能的模式。从 v7.2 开始引入，目前为实验性功能。`OFF` 指的是不开启 Runtime Filter 功能，`LOCAL` 指的开启并使用 LOCAL 模式的 Runtime Filter 功能。
+
+### `tidb_runtime_filter_type`
```
Comment (qiancai) [inline system-variables.md:3696]:
```suggestion
### `tidb_runtime_filter_type` <span class="version-mark">从 v7.2.0 版本开始引入</span>
```

File: system-variables.md [hunk 7]
```diff
@@ -3682,6 +3682,28 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+> **警告：**
+>
+> 该功能为实验特性，不建议在生产环境中使用。
```
Comment (qiancai) [inline system-variables.md:3699]:
```suggestion
> 该功能为实验特性，不建议在生产环境中使用。

```

File: system-variables.md [hunk 8]
```diff
@@ -3682,6 +3682,28 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值: `IN`
+- 可选值: `IN`
```
Comment (qiancai) [inline system-variables.md:3705]:
```suggestion
- 默认值：`IN`
- 可选值：`IN`
```

File: system-variables.md (hunk 5 above)
Comment (elsa0520) [reply system-variables.md:3694]: 因为后面会有一个 专门说明 runtime filter 功能的文档。在这里有点不好解释。之所以现在先把 system variables 这个文档先加上是因为 7.2 带了，所以为了发版先加上这两个变量说明。
后面专门功能文档里会详细解释。

Comment (qiancai) [reply system-variables.md:3694]: 那就先不要说是实验特性。因为实验特性也有人会用。

File: system-variables.md [hunk 9]
```diff
@@ -3682,6 +3682,30 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
 
 - 需要注意的是修改该变量不会对已保存的老数据产生影响，只会对修改变量后的新写入数据使用对应版本格式保存。
 
+### `tidb_runtime_filter_mode`
+> **警告：**
+>
+> 该功能为实验特性，不建议在生产环境中使用。
```
Comment (qiancai) [inline system-variables.md:3688]:
```suggestion
> 当前版本中该变量控制的功能尚未完全生效，请保留默认值。
```

File: system-variables.md [hunk 10]
```diff
@@ -3682,6 +3682,30 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值: `OFF`
+- 可选值: `OFF`, `LOCAL`
+- 该变量用于控制 Runtime Filter 功能的模式。从 v7.2 开始引入，目前为实验性功能。`OFF` 指的是不开启 Runtime Filter 功能，`LOCAL` 指的开启并使用 LOCAL 模式的 Runtime Filter 功能。
+
+### `tidb_runtime_filter_type`
+> **警告：**
+>
+> 该功能为实验特性，不建议在生产环境中使用。
```
Comment (qiancai) [inline system-variables.md:3700]:
```suggestion
> 当前版本中该变量控制的功能尚未完全生效，请保留默认值。
```

File: system-variables.md [hunk 11]
```diff
@@ -3682,6 +3682,30 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值: `OFF`
+- 可选值: `OFF`, `LOCAL`
+- 该变量用于控制 Runtime Filter 功能的模式。从 v7.2 开始引入，目前为实验性功能。`OFF` 指的是不开启 Runtime Filter 功能，`LOCAL` 指的开启并使用 LOCAL 模式的 Runtime Filter 功能。
```
Comment (qiancai) [inline system-variables.md:3695]:
```suggestion
```

File: system-variables.md [hunk 12]
```diff
@@ -3682,6 +3682,30 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+> **警告：**
+>
+> 该功能为实验特性，不建议在生产环境中使用。
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值：`IN`
+- 可选值：`IN`
+- 该变量用于调整 Runtime Filter 的类型。从 v7.2 开始引入。`IN` 指的是 Runtime Filter 的类型为 IN，即使用 IN 类型的谓词作为 Filter。
```
Comment (qiancai) [inline system-variables.md:3707]:
```suggestion
```

File: system-variables.md (hunk 5 above)
Comment (elsa0520) [reply system-variables.md:3694]: 好

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**